genesyscloud/routing_utilization/genesyscloud_routing_utilization_proxy.go

```diff
@@ -0,0 +1,99 @@
+package routing_utilization
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/mypurecloud/platform-client-sdk-go/v130/platformclientv2"
+)
+
+var internalProxy *routingUtilizationProxy
+
+type getRoutingUtilizationFunc func(ctx context.Context, p *routingUtilizationProxy) (*platformclientv2.APIResponse, error)
+type updateRoutingUtilizationFunc func(ctx context.Context, p *routingUtilizationProxy, request *platformclientv2.Utilizationrequest) (*platformclientv2.Utilizationresponse, *platformclientv2.APIResponse, error)
+type deleteRoutingUtilizationFunc func(ctx context.Context, p *routingUtilizationProxy) (*platformclientv2.APIResponse, error)
+
+type updateDirectlyFunc func(ctx context.Context, p *routingUtilizationProxy, d *schema.ResourceData, utilizationRequest []interface{}) (*platformclientv2.APIResponse, error)
+
+type routingUtilizationProxy struct {
+	clientConfig                 *platformclientv2.Configuration
```

Comment:
Dont we want to add caching here on the get and getAll calls

Reply:
There is only one utilization resource per and no getAll functionality, i don't think caching is necessary for this resource